### PR TITLE
Add Matrix Room Onboarding

### PR DESCRIPTION
**Pull request by alyssais**

### CODE REVIEW

**Review by zimbatm — APPROVED** [review]

File: matrix/adding-rooms.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - Existing "Admin" permissions can only be removed by a user for themselves
+- Make sure the room is publicly accessible
+- Add `:nixos.org` room alias
+    - Promote it to the room default
+    - Publish it in the nixos.org room directory
+- Invite mjolnir
+    - and set its power level to "Admin"
+- Tell mjolnir to protect the room
+    - `!mjolnir rooms add <alias>`
+- Promote mjolnir to "Admin"
```
Comment (mweinelt) [inline matrix/adding-rooms.md]: That's already done in L52 and needs to happen before L53.

**Review by zimbatm — APPROVED** [review]

File: matrix/adding-rooms.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - `!mjolnir rooms add <alias>`
+- Upload room avatar
+    - Hash the room id with md5 (like done in the script)
+    - Then `wget https://www.gravatar.com/avatar/${HASH}?d=identicon&f=y&s=2048`
+- Add the room to the relevant space
+- Verify permissions on Mjölnir are all good
+    - `!mjolnir verify`
+- Demote `@admin:nixos.org` to default level
+    - Permissions can be restored via Mjolnir, when required
+
```
Comment (mweinelt) [inline matrix/adding-rooms.md:63]: Spurious newline?

File: matrix/adding-rooms.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+## Tasks
+
+From the `@admin:nixos.org` user, execute the following steps:
+
+- Create new room
+    - or join it, if it already exists
+- If you don't have "Admin" permissions, request them from existing admins
+- Ask existing admins to step down to moderator role
+    - Rooms should be owned by the community, represented by the moderation team, so we can guarantee the room will be maintained going forward
+    - We can probably be more lenient for community project rooms
```
Comment (mweinelt) [inline matrix/adding-rooms.md:45]: Do we all agree on this?

File: matrix/adding-rooms.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    - Community
+        - Community projects
+    - International
+        - Regional channels
+    - Teams
+        - Official Teams
+        - Unclear what they map to
+            - https://nixos.org/community/#governance-teams
+            - https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix
+            - https://github.com/orgs/NixOS/teams
```
Comment (mweinelt) [inline matrix/adding-rooms.md:32]: Can we agree on what is the authoritative mapping?

File: matrix/adding-rooms.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+The `@admin:nixos.org` user is capable of adding an alias that is rooted on its own homeserver, which is what we want for branding reasons.
+
+It is however not a server admin, so we cannot rely on https://github.com/grahamc/matrix-management-tools/blob/main/process-all-rooms.sh, which requires access to the `/_synapse/` Admin API.
+
+We can deal with both new and pre-existing rooms. For pre-existing rooms the existing admins need to hand over that permissions, and demote to moderator, so the room will fall exclusively under community administration.
+
+## Preconditions
+
+- Be joined to `#moderators:nixos.org`
+    - so you can control Mjölnir
```
Comment (mweinelt) [inline matrix/adding-rooms.md]:
```suggestion
- Be joined to `#moderators:nixos.org` with your own account
    - so you can control Mjölnir
```

I still want to know who issues commands to mjolnir, it should not be controlled through the `@admin:nixos.org` account ideally.

File: matrix/adding-rooms.md (hunk 3 above)
Comment (Mic92) [reply matrix/adding-rooms.md:45]: Still unclear to me what this means. Lenient in what aspect? Give more moderation power to the previous admins?

Comment (mweinelt) [reply matrix/adding-rooms.md:45]: Leave community projects with admin permissions to the room, as opposed to asking them to step down to moderator.

Imagine you'd want disko on the nixos space. We'd add Mjölnir as admin and ask you to step down to moderator. Yay/nay?

Comment (alyssais) [reply matrix/adding-rooms.md:45]: The process I've been following in the meantime is to give international rooms the choice of whether they want to stay as admins, and to give community rooms the choice of whether they want to stay as admins, _and_ whether they want mjolnir at all.  I think this is a good balance of the proximity of those types of rooms to the project.

Comment (Mic92) [reply matrix/adding-rooms.md:45]: I think most people want to be at least able to set the topic. Don't know what powerlevel this requires. But guess if they own the project they should also own the channel. If they go rogue than we would probably need to remove the channel as a whole from the NixOS space.

Comment (mweinelt) [reply matrix/adding-rooms.md:45]: The things only admins can do by default are:
- enable room encryption
- change history visibility
- change permissions (modify powerlevel)
- change server acls
- upgrade to a newer room version

Setting the topic, kicking, banning, redacting, inviting, modifying the primary room address is all possible as moderator.

Comment (alyssais) [reply matrix/adding-rooms.md:45]: We can also give moderators extra permissions as required.  For example, I've given a moderator the power to change permissions before (which doesn't let them promote anybody higher than themselves).

**Comment by mweinelt** [comment]
I've been referencing this document on all room management efforts, and I don't think we need to hash it out to the last detail for it to be useful. We can iterate as needed and leave some things up for debate, no problem.

**Pull request merged**